apps/builder/app/shared/pages/use-switch-page.ts

```diff
@@ -94,3 +94,27 @@ export const useSyncPageUrl = () => {
     );
   }, [navigate, page, pageHash]);
 };
+
+/**
+ * Synchronize pageHash with scrolling position
+ */
+export const useHashLinkSync = () => {
+  const pageHash = useStore(selectedPageHashStore);
+
+  useEffect(() => {
+    if (pageHash === "") {
+      return;
+    }
+
+    let elementId = decodeURIComponent(pageHash);
+    if (elementId.startsWith("#")) {
+      elementId = elementId.slice(1);
+    }
+
+    // Try find element to scroll to
+    const element = document.getElementById(elementId);
+    if (element !== null) {
+      element.scrollIntoView();
```

Comment:
Here Im emulating naive browser behavior and remix react router behavior, both has not smooth on hash clicks.
Do we want it to be smooth anyway?

Reply:
Idk, feels nice kinda, the default behavior is a bit dull, since user doesn't realize they are being scrolled when it happens fast, it just looks like a jump

Reply:
Webflow also uses smooth by default